src/domain/event/utils.ts

```diff
@@ -855,3 +884,237 @@ export const getRecurringEventPayload = (
     subEvents,
   };
 };
+
+const SKIP_FIELDS = new Set([
+  'location',
+  'keywords',
+  'audience',
+  'languages',
+  'in_language',
+  'sub_events',
+]);
+
+// Enumerate all the property names of an object recursively.
+function* propertyNames(obj: object): any {
+  for (const name of keys(obj)) {
+    const val = (obj as Record<string, unknown>)[name];
+    if (val instanceof Object && !SKIP_FIELDS.has(name)) {
+      yield* propertyNames(val);
+    }
+    if (val && val !== '') {
+      yield name;
+    }
+  }
+}
+
+export const getEventInfoLanguages = (event: EventFieldsFragment): string[] => {
+  const languages = new Set(ORDERED_EVENT_INFO_LANGUAGES);
+  const foundLanguages = new Set<string>();
+
+  for (const name of propertyNames(event)) {
+    if (foundLanguages.size === languages.size) {
+      break;
+    }
+    if (languages.has(name)) {
+      foundLanguages.add(name);
+    }
+  }
+  return Array.from(foundLanguages);
+};
+
+export const getLocalisedObject = (
+  obj?: Maybe<LocalisedFieldsFragment>,
+  defaultValue = ''
+): MultiLanguageObject => {
+  return reduce(
+    ORDERED_EVENT_INFO_LANGUAGES,
+    (acc, lang) => ({
+      ...acc,
+      [lang]: (obj && (obj as Record<string, unknown>)[lang]) || defaultValue,
```

Comment:
`as Record<string, unknown>` 

Tested if it would work without type casting and it does (or it didn't complain). Is there a reason to uset it?

Reply:
Fixed